corgi/core/models.py

```diff
@@ -347,70 +347,39 @@ def get_related_components(build_ids: QuerySet, sort_field: str) -> QuerySet["Co
         # Else self.builds is an empty QuerySet
         return Component.objects.none()
 
-    @staticmethod
-    def get_product_component_relations(
-        variant_ids: list[str], sort_field="external_system_id", only_errata=False
-    ) -> QuerySet["ProductComponentRelation"]:
-        """Returns unique productcomponentrelations with at least 1 matching variant_id,
-        ordered by sort_field. if only_errata is True, only return PCRs with Type=ERRATA
-        """
-        if variant_ids:
-            query = Q()
-            for variant_id in variant_ids:
-                query |= Q(product_ref__contains=variant_id)
-            result = ProductComponentRelation.objects.filter(query).distinct().order_by(sort_field)
-            if only_errata:
-                result = result.filter(type=ProductComponentRelation.Type.ERRATA)
-            return result
-        # Else self.product_variants is an empty QuerySet
-        return ProductComponentRelation.objects.none()
-
     @property
     def builds(self) -> QuerySet:
-        """Return unique build IDs for errata with variant_ids matching self.product_variants"""
-        return (
-            self.get_product_component_relations(
-                self.product_variants, "build_id", only_errata=True
+        """Returns unique productcomponentrelations with at least 1 matching variant or stream,
+        ordered by build_id.
+        """
+        product_refs = []
+        if isinstance(self, ProductVariant):
+            product_refs.append(self.name)
+        elif isinstance(self, ProductStream):
+            product_refs.append(self.name)
+            # we also want to include child product_variants for each product stream in the query
+            product_refs.extend(self.product_variants)
+        if product_refs:
+            return (
+                ProductComponentRelation.objects.filter(product_ref__in=product_refs)
```

Comment:
This does an exact lookup, where the product_ref stored in the PCR table has to exactly match one of the variant names in the product_refs list / self.product_variants field.

The old get_pcr method (which I refactored slightly without changing logic) did a lookup to check if each variant name was inside of / a substring of the product_ref. Is that substring check unneeded, AKA do we always store exactly the variant name as the product_ref?